Git/config.sublime-syntax

```diff
@@ -0,0 +1,206 @@
+      scope: invalid.illegal.gitconfig
+
+  illegal-slash:
+    - match: \\
+      scope: invalid.illegal.gitconfig
+
+  # changed = red
+  # untracked = bold green
+  key-color-pair:
+    - match: ^\s*({{variable_name}})\s*(\=)
+      captures:
+        1: variable.other.readwrite.gitconfig
+        2: keyword.operator.assignment.gitconfig
+      push:
+        - meta_scope: meta.key-value-pair.gitconfig
+        - meta_content_scope: meta.value.gitconfig
+        - match: '(?=^\s*{{variable_name}}\s*\=)'  # next key-value pair
+          pop: true
+        - match: (?=^\s*\[)                        # start of new section
+          pop: true
+        - include: color
+
+  # key = val
+  key-value-pair:
+    - match: ^\s*({{variable_name}})\s*(\=)
+      captures:
+        1: variable.other.readwrite.gitconfig
+        2: keyword.operator.assignment.gitconfig
+      push:
+        - meta_scope: meta.key-value-pair.gitconfig
+        - meta_content_scope: meta.value.gitconfig
+        - match: '(?=^\s*{{variable_name}}\s*\=)'  # next key-value pair
+          pop: true
+        - match: (?=^\s*\[)                        # start of new section
+          pop: true
+        - include: value
+
+  # [section "subsection"]
+  section-header:
+    - match: \[
+      scope: punctuation.definition.brackets.begin.gitconfig
+      push:
+        - meta_scope: meta.brackets.gitconfig
+        - match: \]
+          scope: punctuation.definition.brackets.end.gitconfig
+          pop: true
+        - include: illegal-newline  # complain if '[' and ']' on separate lines
+        - include: illegal-slash
+        - include: subsection-name
+        - include: section-name
+    - include: stray-bracket
+
+  # A section is a [section header] and the list of key-value pairs under it.
+  #
+  # [section "subsection"]
+  #   key1 = value1
+  #   key2 = value2
+  section:
+    # [color] subsections are matched separately, because they are the only
+    # sections that take colors for values.
+    - include: section-color  # e.g. [color "branch"], [color "status"]
+    - include: section-other
+
+  # We only want to match color values under these subsections:
+  #   [color "branch"]
+  #   [color "diff"]
+  #   [color "interactive"]
+  #   [color "status"]
+  # Note that this does not match [color] (without the subsection). This is
+  # because [color] doesn't actually take color values.
+  section-color:
+    - match: ^\s*(?=\[\s*color\s+\")
+      push:
+        - match: (?=^\s*\[) # start of new section
+          pop: true
+        - include: section-header
+        - include: key-color-pair
+
+  section-name:
+    - match: '{{section_name}}'
+      scope: entity.name.gitconfig
+
+  # section-other matches all sections except [color "subsection"].
+  section-other:
+    - match: ^\s*(?=\[)
+      push:
+        - match: (?=^\s*\[) # start of new section
+          pop: true
+        - include: section-header
+        - include: key-value-pair
+
+  stray-bracket:
```

Comment:
For code, having functions for self-documenting code is usually good practice, but as you mentioned, the scope name is very self-documenting in syntax definitions as well, so that doesn't really hold here.